docs/rfcs/rfc-007-subscriptions.md

```diff
@@ -209,39 +330,253 @@ class Neo4jGraphQLSubscriptionsAMQPPlugin extends Neo4jGraphQLSubscriptionsPlugi
+  updateMovies(where: {title: "Cornetto Movie"}, update: {title: "The Matrix"}) {
+    movies {
+      title
+    }
+  }
+}
 ```
 
+To support use cases for subscriptions to an updated value, update subscriptions should also provide filters targeting the new value by prepending `NEW`, for instance:
+
+* `NEW_title`.
+* `NEW_title_GT`.
+
+This would allow for subscriptions such as:
+```graphql
+subscription SubscribeToConfirmedUsers {
+  userUpdated(where: { AND: [{status: "pending"}, {NEW_status: "confirmed"}] }) {
+    name
+    status
+  }
+}
+```
+
+### Auth
+A new operation `SUBSCRIBE` to deal with authorization on subscriptions, this operation will follow the same syntax and rules as any other operation, but with more
+limited scope, allowing only for top-level properties validation (same limitations as `where`).
+
+#### Authentication
+```graphql
+type Post @auth(rules: [
+    { operations: [SUBSCRIBE], isAuthenticated: true }
+]) {
+    title: String!
+}
+```
+
+If the user is not authenticated, the subscription request will fail.
+
+#### Allow
+Works as usual, but not supported throughout relationships:
+
+```graphql
+type User @auth(
+    rules: [
+        {
+            operations: [SUBSCRIBE],
+            allow: { id: "$jwt.sub" }
+        }
+    ]
+) {
+    id: ID!
+    name: String!
+}
+```
+
+This requires `jwt` to be available in the context on subscription resolver. If a rule does not match, the subscription will not be sent.
+
+#### Bind
+Bind does not make sense on subscriptions, as it is intended for mutations. So bind is ignored for SUBSCRIBE operations
+
+#### Roles
+Roles should work as usual, the subscription request should error if the role does not match.
+
+#### Where
+Where filter in the auth directive should work under the same limitations and behaviour as the subscriptions filtering. With any subscription event not matching
+this where not being sent.
+
+### Extra fields in subscription
+The following should be available along with the data payload:
+- **event**: The event that triggered this event (`CREATE`, `UPDATE`, `DELETE`)
+- **timestamp**: The timestamp the event was generated, this may be needed to ensure order consistency on horizontal scaled subscriptions.
+- **previousState** (only for updates): State of the node before the modification.
+
+
+### Nested fields
+Due to subscriptions relying on the `meta` data returned by the queries, only top-level properties of the node are available to return or filter.
+
+To properly support nested queries, follow-up queries to the database, after the subscription has been triggered would be required. This may be not viable as each subscription would require an extra request, this not only put a huge load to the database, the subscription itself will take longer to fulfill, instead of close to real-time. Adding some extra metadata and caching subscriptions may help,
+
+Due to this, any nested field queries support should be opt-in.
+
+### Horizontal scalability
```

Comment:
(It's not this line)
Just want to remind ourselves to think about any security considerations. For instance "How can a malicious user abuse this?" and in such a case even "How could we prevent that?"

Reply:
Great point, I'll add a new section covering any security consideration